File: slicing-android/app/src/main/java/com/crescentflare/slicinggame/page/modules/PageModule.kt
package com.crescentflare.slicinggame.page.modules

import com.crescentflare.slicinggame.infrastructure.events.AppEvent
import com.crescentflare.slicinggame.page.activities.PageActivity

/**
 * Page module: provides an interface for separating controller logic into modules
 */
interface PageModule {

    val handleEventTypes: List<String>

    fun onCreate(activity: PageActivity)
    fun onResume()
    fun onPause()
    fun onDestroy()
    fun onLowMemory()
    fun onBackPressed(): Boolean

    fun handleEvent(event: AppEvent, sender: Any?): Boolean

}

File: slicing-ios/SlicingGame/Page/Modules/PageModule.swift
//
//  PageModule.swift
//  Page module: provides a protocol for separating controller logic into modules
//

protocol PageModule: class {
    
    var handleEventTypes: [String] { get }
    
    func didCreate(viewController: PageViewController)
    func didResume()
    func didPause()
    func willDestroy()

    func handleEvent(_ event: AppEvent, sender: Any?) -> Bool

}
